src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,44 @@
-import React from 'react';
+/* eslint-disable no-console */
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+import cn from 'classnames';
+
+type Props = {
+  handleTodoReset: (value: Todo | null) => void;
+  selectedUserId: number;
+  selectedTodo: Todo | null;
+};
+
+const defaultUser: User = {
+  id: 0,
+  name: 'Default Username',
+  email: 'Default Email',
+  phone: 'Default Phonenumber',
+};
+
+export const TodoModal: React.FC<Props> = ({
+  handleTodoReset,
+  selectedUserId,
+  selectedTodo,
+}) => {
+  const [user, setUser] = useState<User>(defaultUser);
```

Comment:
You are initializing state with a `defaultUser` object. This can lead to bugs and unexpected behavior because the state will not update when the props change. Consider a different approach to handle initial state. You can initialize the state to `null` and handle the loading and error states separately.

Reply:
....